File: FrontEnd/src/app/models/DonDTO.ts
import { DemandeDTO } from "./DemandeDTO";

export interface DonDTO {
    id: number;
    montant: string;
    dateRemise: string;
    type: string;
    campagne: any; // ou `CampagneDTO` si tu le définis
    demandes: DemandeDTO[];
  }

File: donation/src/main/java/com/example/donation/entity/DonDTO.java
package com.example.donation.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class DonDTO {
    private Long id;
    private String montant;
    private LocalDate dateRemise;
    private String type;
    private CampagneRequestDTO campagne;
    private List<DemandeDTO> demandes;
}
